### PR TITLE
[workload-testing] Update testing nuget, and remove some workarounds

### PR DESCRIPTION
**Pull request by radical**
Contributes to e2e follow up work - https://github.com/dotnet/aspire/issues/2687 .

`Update Microsoft.NET.Runtime.WorkloadTesting.Internal to 9.0.0-preview.3.24158.8`

```
[workload-testing] Remove the hacks, and use the update task

  1. remove package source mapping from `nuget.config` as it gets added at
     build time now
  2. Manifest packages are installed with a `PackageDownload` instead of a
     `PackageReference`, thus it can work with
     `PackageType='dotnetplatform'` packages.
  3. Generating a patched nuget.config for building test projects can be
     done using a new `PatchNuGetConfig` task, and thus we can remove the
     inline task.
```
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2748)

### CODE REVIEW

File: tests/Shared/Aspire.Workload.Testing.targets [hunk 1]
```diff
@@ -3,6 +3,11 @@
     <GetWorkloadInputsDependsOn>_GetWorkloadsToInstall;$(GetWorkloadInputsDependsOn)</GetWorkloadInputsDependsOn>
     <GetNuGetsToBuildForWorkloadTestingDependsOn>_GetNuGetsToBuild;$(GetNuGetsToBuildForWorkloadTestingDependsOn)</GetNuGetsToBuildForWorkloadTestingDependsOn>
 
+    <!-- this will cause the output for `dotnet workload install` to be visible to the user -->
+    <WorkloadInstallCommandOutputImportance>High</WorkloadInstallCommandOutputImportance>
+
```
Comment (radical) [inline tests/Shared/Aspire.Workload.Testing.targets:8]:
```suggestion
```

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -73,9 +73,9 @@
       <Uri>https://dev.azure.com/dnceng/internal/_git/dotnet-runtime</Uri>
       <Sha>5535e31a712343a63f5d7d796cd874e563e5ac14</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NET.Runtime.WorkloadTesting.Internal" Version="9.0.0-preview.3.24151.4">
+    <Dependency Name="Microsoft.NET.Runtime.WorkloadTesting.Internal" Version="9.0.0-preview.3.24158.8">
       <Uri>https://github.com/dotnet/runtime</Uri>
-      <Sha>50d6e5d5ffd05dd4034cffd222ea610baedcc326</Sha>
+      <Sha>54d318d6da76e409d4e865220d9923283d55a06d</Sha>
```
Comment (eerhardt) [inline eng/Version.Details.xml:78]: Do we have this set up on a maestro auto-update?

Comment (radical) [reply eng/Version.Details.xml:78]: After this PR, I can set that up. I wanted to get the latest changes in first.

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**